packages/shared/components/MenuLogin/MenuLogin.test.tsx

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import { render, fireEvent, waitFor } from 'design/utils/testing';
+import { MenuLogin } from './MenuLogin';
+
+test('does not accept an empty value when required is set to true', async () => {
+  const onSelect = jest.fn();
+  const { findByText, findByPlaceholderText } = render(
+    <MenuLogin
+      placeholder="MenuLogin input"
+      required={true}
+      getLoginItems={() => []}
+      onSelect={() => onSelect()}
+    />
+  );
+
+  fireEvent.click(await findByText('CONNECT'));
+  await waitFor(async () =>
+    fireEvent.keyPress(await findByPlaceholderText('MenuLogin input'), {
+      key: 'Enter',
+      keyCode: 13,
+    })
+  );
+
+  expect(onSelect).toHaveBeenCalledTimes(0);
```

Comment:
I wasn't sure whether to put `waitFor` on `fireEvent` vs `expect().toHaveBeenCalledTimes(0)`.

My understanding was that in order to verify that something was not at all, `waitFor` would have to wait until its default timeout runs out. To verify if `fireEvent` was called, it only needs to check if the event was fired.
 
Turns out this is not how `waitFor` works at all. If the callback doesn't return a promise, `waitFor` just continuously calls the callback until it stops throwing an error or the timeout runs out.

Anyway, I guess it's more readable to have expectations without `waitFor`.